jbcli/jbcli/utils/dockerutil.py

```diff
@@ -37,27 +37,29 @@ def on_modified(self, event):
             path = re.split(r'[\\/]', event.src_path)
         else:
             path = event.src_path.split('/')
-        click.echo('Change detected in app: {}.'.format(event.src_path))
-
-        if '.git' not in path and 'builds' not in path:
-            file_parts = path[-1].split('.')
-            if len(file_parts) == 2:
-                extension = path[-1].split('.')[-1]
-                is_py = extension == 'py'
-                if not is_py:
-                    # Path looks like 
-                    # ['..', '..', 'apps', 'privileging', 'stacks', 'overview', 'templates.html']
-                    app = path[3]
-                    if not load_app(app):
-                        run('/venv/bin/python manage.py loadjuiceboxapp {}'.format(
+
+        # Path looks like 
+        # ['..', '..', 'apps', 'privileging', 'stacks', 'overview', 'templates.html']
+        app = path[3]
+        filename = path[-1]
+        is_python_change = filename.endswith('.py') and isinstance(event, FileModifiedEvent)
+
+        if 'builds' not in path and '.idea' not in path and '.git' not in path:
+            click.echo('Change detected in {}.'.format(app))
+            if is_python_change:
+                # We don't need to reload the app just refresh
+                # the browser after juicebox service restarts
+                refresh_browser(5)
+            else:
+                # Try to load app via api, fall back to calling docker.exec_run
+                if not load_app(app):
+                    run('/venv/bin/python manage.py loadjuiceboxapp {}'.format(
                                 app))
-                    echo_success('{} was added successfully.'.format(app))
+                echo_success('{} was added successfully.'.format(app))
+                refresh_browser()
 
-                if self.should_reload:
```

Comment:
I think the use of `self.should_reload` may have gotten lost in the shuffle, but I think we still need to refer to this to determine whether or not the `--reload` flag was used when `jb watch` was started.

Reply:
Thanks